kernel/include/kernel_structs.h

```diff
@@ -89,6 +88,30 @@ struct _kernel {
 
 	/* currently scheduled thread */
 	struct k_thread *current;
+};
+
+typedef struct _cpu _cpu_t;
+
+struct _kernel {
+	/* For compatibility with pre-SMP code, union the first CPU
+	 * record with the legacy fields so code can continue to use
+	 * the "_kernel.XXX" expressions and assembly offsets.
+	 */
+	union {
+		struct _cpu cpus[CONFIG_MP_NUM_CPUS];
```

Comment:
strcut _cpu should be under #ifdef CONFIG_SMP right?

Reply:
No, this is the official API now.  When there is only one CPU, the array is of length one but otherwise shares the same offsets as the legacy fields in the union (which will eventually go away).  There's no particular benefit to having a separate API beyond making migration easier.